From 6399c77e5a71ac2cb954ac695f3536ed6b7c94b6 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 24 Apr 2019 10:08:43 +0300
Subject: [PATCH] [SW-0] pll: override slip wd counter

The counter counts the cycles a PLL was in an unlocked state and
produces an interrupt on counter overlap.
Program the counter to 1 cycle to produce an interrupt each time
a PLL lock is lost.

Change-Id: Ie682c72708c88b26f3e07a9abbb9af2bf1fb37bd
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index a4f1dba..977a490 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -79,6 +79,9 @@
 
 	hl_pll_reset(cfg, false);
 
+	/* configure to raise interrupt after X unlocked cycles */
+	regmap_writel(0x1, cfg->base + HL_PLL_SLIP_WD_CNTR);
+
 	/* Wait >500 (10us) reference cycles until PLL re-locks. */
 	hl_pll_udelay(HL_PLL_MAX_WAIT_TIME);
 
diff --git a/common/pll.h b/common/pll.h
index 15c74c7..17221a5 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -17,6 +17,7 @@
 #define HL_PLL_LOCK_INTR	0x128
 #define HL_PLL_DATA_CHNG	0x130
 #define HL_PLL_RST		0x134
+#define HL_PLL_SLIP_WD_CNTR	0x150
 #define HL_PLL_DIV_FACTOR	0x200
 #define HL_PLL_DIV_FACTOR_CMD	0x220
 #define HL_PLL_DIV_SEL		0x280
